Upstream-Status: Inappropriate [embedded specific: only for testing]

This patch allows Chromium to be run as root by explicitly setting the root
profile directory to /root/chromium-profile, otherwise Chromium refuses to
launch.

DO NOT USE THIS IN PRODUCTION; this patch is only supposed to ease testing and
debugging.

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
diff --git a/chrome/installer/linux/common/wrapper b/chrome/installer/linux/common/wrapper
index 83c2ba4..ea42ea0 100755
--- a/chrome/installer/linux/common/wrapper
+++ b/chrome/installer/linux/common/wrapper
@@ -45,6 +45,15 @@ exec < /dev/null
 exec > >(exec cat)
 exec 2> >(exec cat >&2)
 
+# Specify a profile directory when running as root, otherwise Chromium will
+# refuse to launch.
+# DO NOT USE IN PRODUCTION! This change is only supposed to help testing your
+# deployment when everything is run as root (think core-image-sato, for
+# instance).
+if [ `id -u` -eq "0" ]; then
+    export CHROME_USER_DATA_DIR="/root/chromium-profile"
+fi
+
 # Make sure that the profile directory specified in the environment, if any,
 # overrides the default.
 if [[ -n "$CHROME_USER_DATA_DIR" ]]; then
 